src/coreclr/tools/aot/Mono.Linker.Tests/TestCasesRunner/ResultChecker.cs

```diff
@@ -220,8 +214,9 @@ private void VerifyLoggedMessages (AssemblyDefinition original, TestLogWriter lo
 											loggedMessages.Remove (loggedMessage);
 											break;
 										}
-										if (actualName.StartsWith (expectedMember.DeclaringType.FullName) &&
-											actualName.Contains (".cctor") && (expectedMember is FieldDefinition || expectedMember is PropertyDefinition)) {
+										if (actualName.StartsWith ($"[test]{expectedMember.DeclaringType.FullName.Replace ("/", ".")}") &&
+											actualName.Contains (".cctor") &&
+											(expectedMember is FieldDefinition || expectedMember is PropertyDefinition)) {
```

Comment:
Ideally we should rewrite this to use GetExpectedOriginDisplayName and GetActualOriginDisplayName - look at LogMessageHasSameOriginMember where we compare Cecil and Internal.TypeSystem members. We basically rely on the display name which is currently "C# like" and on the code on both sides to adapt the specific type systems to it.

It tends to work pretty well.